OpenSim/Simulation/Model/Model.h

```diff
@@ -254,6 +256,12 @@ OpenSim_DECLARE_CONCRETE_OBJECT(Model, ModelComponent);
      */
     void cleanup();
 
+    /** Model clone() override that invokes finalizeFromProperties() 
+        on a default copy constructed Model, prior to returning the Model. */
+    Model* clone() const override;
+    
+    const std::string& getConcreteClassName() const override
+    {   return getClassName(); }
```

Comment:
Consider adding a comment like "This function is usually defined via `OpenSim_DECLARE_CONCRETE_OBJECT()`, but we do not use that macro for Model so that we can define a custom `clone()`. This comment could go here or at the top of the class where the macro usually appears.